File: aoc2023/test/aoc2023/day02_test.clj
(ns aoc2023.day02-test
  (:require [clojure.test :refer :all]
            [clojure.java.io]
            [aoc2023.day02 :as day01]))

(def sample-input (slurp (clojure.java.io/resource "day02-sample.txt")))

(deftest test-part1
  (testing "Day 2 Part 1"
    (is (= 8 (day01/part1 sample-input)))
  )
)

(deftest test-part2
  (testing "Day 2 Part 2"
    (is (= 2286 (day01/part2 sample-input)))
  )
)

File: 2024/day02/day02_test.go
package day02

import (
	"fmt"
    "testing"
    )

var sample = "sample.txt"

func TestPart1(t *testing.T) {
  result := Part1(sample)
  if result != 2 {
    fmt.Println(result)
  	t.Fail()
  }
}

func TestPart2(t *testing.T) {
  result := Part2(sample)
  // not 290
  if result != 4 {
    fmt.Println(result)
  	t.Fail()
  }
}
